File: app/src/main/java/com/example/administrator/shadowapplication/thread_test/AsyncTaskActivity.java
package com.example.administrator.shadowapplication.thread_test;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.example.administrator.shadowapplication.R;

public class AsyncTaskActivity extends AppCompatActivity {

    /**
     * 测试异步线程处理数据
     *
     * @param savedInstanceState
     */
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_async_task);
        Button startTask = (Button) findViewById(R.id.btn_start_task);
        startTask.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                new AsyncTaskSelf().execute();
            }
        });
    }
}

File: app/src/main/java/com/example/administrator/shadowapplication/handle_msg/AsyncTaskActivity.kt
package com.example.administrator.shadowapplication.handle_msg

import android.os.AsyncTask
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import com.example.administrator.shadowapplication.R
import java.text.SimpleDateFormat
import java.util.*

class AsyncTaskActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_async_task2)
        MyAsyncTask("syncTask1").execute()
        MyAsyncTask("syncTask2").execute()
        MyAsyncTask("syncTask3").execute()
        MyAsyncTask("syncTask4").execute()
       /* mAsync.execute()
        mAsync.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR,"")*/
    }

    class MyAsyncTask constructor(var name:String):AsyncTask<String, Int, Boolean>(){

        override fun onPreExecute() {
            super.onPreExecute()
            Log.d("hh","onPreExecute $name")
        }

        override fun doInBackground(vararg params: String?): Boolean {
            Thread.sleep(300)
            return true
        }


        override fun onPostExecute(result: Boolean?) {
           val df =  SimpleDateFormat("yyyy-MM-dd HH:mm:ss")
            Log.d("hh","onPostExecute $name =  ${df.format(Date())}")
        }

    }
}
